Error compiling AESNI in Mbed-TLS with clang on Windows

### Summary

clang does not have MSVS intrinsics, the pre-processor condition below does not cover this case if clang runs in MSVC compatibility mode
https://github.com/Mbed-TLS/mbedtls/blob/1ec69067fa1351427f904362c1221b31538c8b57/library/aesni.h#L42-L46
It should be
```c
#if defined(_MSC_VER) && !defined(__clang__)
/* Visual Studio supports AESNI intrinsics since VS 2008 SP1. We only support
 * VS 2013 and up for other reasons anyway, so no need to check the version. */
#define MBEDTLS_AESNI_HAVE_INTRINSICS
#endif
```

### System information

Mbed TLS version 2.28.5, 3.5.0
Operating system and version: Windows 10/11
Default configuration.
Compiler: LLVM clang, clang-cl

### Expected behavior

Built without errors.

### Actual behavior

Got many errors
```
/mbedtls/library/aesni.c:62:17: error: passing 'unsigned int[4]' to parameter of type 'int *' converts between pointers to integer types with different sign [-Werror,-Wpointer-sign]
        __cpuid(info, 1);
                ^~~~
C:\PROGRA~1\LLVM\lib\clang\16\include\intrin.h:60:17: note: passing argument to parameter here
void __cpuid(int[4], int);
                ^
/mbedtls/library/aesni.c:102:21: error: call to undeclared function '_mm_aesdec_si128'; ISO C99 and later do not support implicit function declarations [-Wimplicit-function-declaration]
            state = _mm_aesdec_si128(state, *rk);
                    ^
/mbedtls/library/aesni.c:102:21: note: did you mean '_mm_and_si128'?
C:\PROGRA~1\LLVM\lib\clang\16\include\emmintrin.h:2611:46: note: '_mm_and_si128' declared here
static __inline__ __m128i __DEFAULT_FN_ATTRS _mm_and_si128(__m128i __a,
                                             ^
<...>
```

### Additional infromation

Fixing PRs will be offered soon.

--- a/ChangeLog.d/8372.txt
+++ b/ChangeLog.d/8372.txt
@@ -1,0 +1,3 @@
+Features
+   *  AES-NI is now supported in Windows builds with clang and clang-cl.
+      Resolves #8372.

--- a/include/mbedtls/aesni.h
+++ b/include/mbedtls/aesni.h
@@ -58,7 +58,7 @@
  *       macros that may change in future releases.
  */
 #undef MBEDTLS_AESNI_HAVE_INTRINSICS
-#if defined(_MSC_VER)
+#if defined(_MSC_VER) && !defined(__clang__)
 /* Visual Studio supports AESNI intrinsics since VS 2008 SP1. We only support
  * VS 2013 and up for other reasons anyway, so no need to check the version. */
 #define MBEDTLS_AESNI_HAVE_INTRINSICS
@@ -66,7 +66,7 @@
 /* GCC-like compilers: currently, we only support intrinsics if the requisite
  * target flag is enabled when building the library (e.g. `gcc -mpclmul -msse2`
  * or `clang -maes -mpclmul`). */
-#if defined(__GNUC__) && defined(__AES__) && defined(__PCLMUL__)
+#if (defined(__GNUC__) || defined(__clang__)) && defined(__AES__) && defined(__PCLMUL__)
 #define MBEDTLS_AESNI_HAVE_INTRINSICS
 #endif
 

--- a/library/aesni.c
+++ b/library/aesni.c
@@ -57,7 +57,7 @@ int mbedtls_aesni_has_support(unsigned int what)
 
     if (!done) {
 #if MBEDTLS_AESNI_HAVE_CODE == 2
-        static unsigned info[4] = { 0, 0, 0, 0 };
+        static int info[4] = { 0, 0, 0, 0 };
 #if defined(_MSC_VER)
         __cpuid(info, 1);
 #else
@@ -191,7 +191,7 @@ void mbedtls_aesni_gcm_mult(unsigned char c[16],
                             const unsigned char a[16],
                             const unsigned char b[16])
 {
-    __m128i aa, bb, cc, dd;
+    __m128i aa = { 0 }, bb = { 0 }, cc, dd;
 
     /* The inputs are in big-endian order, so byte-reverse them */
     for (size_t i = 0; i < 16; i++) {

